src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -496,8 +516,26 @@ export class ReplaceAllAction extends AbstractSearchAndReplaceAction {
 
 	override run(): Promise<any> {
 		const tree = this.viewlet.getControl();
-		const nextFocusElement = this.getElementToFocusAfterRemoved(tree, this.fileMatch);
-		return this.fileMatch.parent().replace(this.fileMatch).then(() => {
+
+		let elementsToReplace: RenderableMatch[] = [];
+		const currentSelection: RenderableMatch[] = [];
+
+		this.viewlet.getControl().getSelection().forEach((elem) => {
+			currentSelection.push(<RenderableMatch>elem);
+		});
+
+		// if you're using multiselect and you click to replace one element from the tree, replace all selected
+		if (currentSelection.length > 1 && currentSelection.includes(this.fileMatch)) {
+			elementsToReplace = currentSelection;
+		} else {
+			elementsToReplace.push(this.fileMatch);
+		}
+		const currFocusElement = elementsToReplace[elementsToReplace.length - 1];
+		const nextFocusElement = this.getElementToFocusAfterRemoved(tree, currFocusElement);
+
+		return Promise.all(elementsToReplace.map(async (elem) => {
+			await elem.parent().replace(<any>elem);
```

Comment:
Why the cast to `any`?

Reply:
update: the reason I used `<any>` is because this call to `replace` required something of `never` type

Reply:
That's a sign that something funky is going on. Having to cast to any is a red flag to me